src/func_surf_energy_bal.c

```diff
@@ -687,15 +690,65 @@ double func_surf_energy_bal(double Ts, va_list ap)
 			    TMean, Tair, wind[UnderStory], roughness[UnderStory]);
   else
     Ra_used[0] = HUGE_RESIST;
-  
+
+  /*************************************************
+    Compute aerodynamic resistance for the case of exposed soil between
+    plants (or gaps in canopy).  Assume plants and exposed soil are well-
+    mixed, i.e., exposed soil is neither as disconnected from the
+    atmosphere as soil under veg or canopy, nor as exposed as a large
+    area of exposed soil.  Rather, it is subject to some wind attenuation
+    from the surrounding plants.  Thus, compute as the area-weighted
+    average of "pure" understory and "pure" exposed conditions.
+
+    NOTE: can't average the resistances; must convert to conductances,
+    average the conductances, and then convert the average conductance
+    to a resistance.
+  *************************************************/
+  if (veg_var->vegcover < 1) {
+    /** If Ra_used (which pertains to under vegetation at this point)
+        is non-zero, use it to compute area-weighted average **/
+    if (Ra_used[0] > 0) {
+      /** aerodynamic conductance under vegetation **/
+      ga_veg = 1/Ra_used[0];
+      /** compute aerodynamic resistance over exposed soil (Ra_bare) **/
+      tmp_wind[0] = wind[0];
+      tmp_wind[1] = -999.; // unused
+      tmp_wind[2] = -999.; // unused
+      tmp_height = soil_con->rough/0.123; // eqn from calc_veg_roughness()
+      tmp_displacement[0] = calc_veg_displacement(tmp_height);
+      tmp_roughness[0] = soil_con->rough;
+      tmp_ref_height[0] = 10; // wind measurement height over bare soil
+      Error = CalcAerodynamic(0,0,0,soil_con->snow_rough,soil_con->rough,0,Ra_bare,tmp_wind,tmp_displacement,tmp_ref_height,tmp_roughness);
+      Ra_bare[0] /= StabilityCorrection(tmp_ref_height[0], tmp_displacement[0], TMean, Tair, tmp_wind[0], tmp_roughness[0]);
+      /** if Ra_bare is non-zero, compute area-weighted average
+          aerodynamic conductance **/
+      if (Ra_bare[0] > 0) {
+        /** aerodynamic conductance over exposed soil **/
+        ga_bare = 1/Ra_bare[0];
+        /** area-weighted average aerodynamic conductance **/
+        ga_average = veg_var->vegcover*ga_veg + (1-veg_var->vegcover)*ga_bare;
+        /** aerodynamic resistance is inverse of conductance **/
+        Ra_used[0] = 1/ga_average;
+      }
+      /** else aerodynamic resistance is zero **/
+      else {
+        Ra_used[0] = 0;
```

Comment:
It should be zero.  Ra of 0 means infinite conductance.  Perhaps some of the confusion here is due to the naming of the variables.  Before I added this feature, Ra_used was the Ra of the entire veg tile.  Since vegcover was assumed == 1.0 in the past, Ra_used was also the Ra of the vegetated portion (which was 100% of the tile).  When I added this block, I intended to adjust Ra_used for the presence of exposed soil.  In this case, the old Ra_used now initially only pertains to the vegetated portion (which is < 100% of the tile), and gets reassigned to the value pertaining to the entire tile later on.

So, what we have is:
If vegetated Ra is 0, then vegetated conductance (ga) is infinite, so total tile ga is also infinite, so Ra_used = 1/ga is 0.
If exposed soil Ra is 0, Ra_used also = 0.

What if I add another intermediate variable to make this clearer: Ra_veg.  Ra_veg will be assigned = Ra_used at the beginning of this block.  Then, ga_veg = 1/Ra_veg.  ga_avg = weighted average as currently stated.  Then, Ra_used = 1/ga_avg.  This will hopefully make it clearer that Ra_used = 0 if either Ra_veg or Ra_bare are 0.